framework/framework.go

```diff
@@ -126,6 +145,29 @@ func (f *Framework) AddFunc(obj interface{}) {
 	f.logger.Log("action", "end", "component", "operatorkit", "function", "ProcessCreate")
 }
 
+func (f *Framework) Boot() {
+	f.bootOnce.Do(func() {
+		operation := func() error {
+			err := f.bootWithError()
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		}
+
+		notifier := func(err error, d time.Duration) {
+			f.logger.Log("warning", fmt.Sprintf("retrying operator boot due to error: %#v", microerror.Mask(err)))
+		}
+
+		err := backoff.RetryNotify(operation, f.backOffFactory(), notifier)
+		if err != nil {
+			f.logger.Log("error", fmt.Sprintf("stop operator boot retries due to too many errors: %#v", microerror.Mask(err)))
+			os.Exit(1)
```

Comment:
This is just copy/paste. We did it like this in the operators so far. The backoff factory returns a backoff which is used for the boot and if the boot fails a couple of times the operator dies.

Reply:
I think it'd be nice if Boot had context argument and returned error.

Reply:
That would change the implementation as it is right now. I would like to go with what we have and sort a better boot mechanism separately.